src/Compiler/Utilities/range.fs

```diff
@@ -258,6 +259,13 @@ module FileIndex =
     let startupFileName = "startup"
     let commandLineArgsFileName = "commandLineArgs"
 
+    let mutable internal testSource: string option = None
+
+    let internal setTestSource source =
+        testSource <- Some source
+        { new IDisposable with
```

Comment:
Could we maybe make the source accessible from this object? So that we can pass it to `parseTestSource` directly in the tests. That would make the test code a bit clearer and if we ever need to change how we store this we wouldn't need to change the tests.